org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextFormatterJava.java

```diff
@@ -202,13 +209,19 @@ private void formatGroupElements(CompoundElement group, IFormattableDocument doc
 		boolean first = true;
 		EList<AbstractElement> elements = group.getElements();
 		int size = elements.size();
-		int index = 0;
-		for (AbstractElement element : elements) {
-			index++;
-			boolean last = index == size;
+		for (int index = 0; index < elements.size(); index++) {
+			boolean last = index + 1 == size;
+			AbstractElement element = elements.get(index);
+			AbstractElement next = null;
+			if (!last)
+				next = elements.get(index + 1);
+			if (first && !last) {
+				if (!(element instanceof UntilToken) && !(element instanceof Group) && !(next instanceof UntilToken))
```

Comment:
Can we extract this if condition and give it a name?

Reply:
Extracted a method.